lib/socket-pool.js

```diff
@@ -0,0 +1,181 @@
+const udx = require('udx-native')
+
+module.exports = class SocketPool {
+  constructor (dht) {
+    this._dht = dht
+
+    this._sockets = new Set()
+    this._free = new Set()
+    this._pairers = new Map()
+    this._ids = new Set()
+
+    // Indexes
+    this._byPort = new Map()
```

Comment:
don't think we need a port index actually, but we can keep it until we know for sure

Reply:
We shouldn't need one, no.